### PR TITLE
yt-dlp: 2021.08.02 -> 2021.08.10

### PR DESCRIPTION
**Pull request by kidonng**
###### Motivation for this change

https://github.com/yt-dlp/yt-dlp/releases/tag/2021.08.10

This also switches to use the pypi package which

- Fix wrong version number in the program (the version is bumped after the git tag)
- Brings shell completions
- Brings manpage

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/yt-dlp/default.nix [hunk 1]
```diff
@@ -10,18 +9,20 @@
...
-  src = fetchFromGitHub {
-    owner = "yt-dlp";
-    repo = "yt-dlp";
-    rev = version;
-    sha256 = "QEJKOZGVQNXLU8GfTbwBx2Zv3KO++ozTJcXLWxXA4hI=";
+  version = "2021.08.10";
+
+  src = fetchPypi {
+    inherit pname;
+    version = with lib; concatMapStringsSep "." (removePrefix "0") (with versions; [(major version) (minor version) (patch version)]);
```
Comment (kidonng) [inline pkgs/tools/misc/yt-dlp/default.nix:16]: I don't know a better way to remove the leading zeros from package version, suggestions welcome.

Comment (SuperSandro2000) [reply pkgs/tools/misc/yt-dlp/default.nix:16]: We cna also just change the version number.

Comment (kidonng) [reply pkgs/tools/misc/yt-dlp/default.nix:16]: The one without leading zeros seems to be only used for pypi. This just looks not very nice but shouldn't be a real trouble.

youtube-dl is in the same fashion (program version is `2021.06.06` while pypi is `2021.6.6`), however their version is "normalized" by some certain script (I didn't get to remember which one it is, may try again later).

Comment (mkg20001) [reply pkgs/tools/misc/yt-dlp/default.nix:16]: Another way would be to do

```nix
builtins.replaceStrings [ ".0" ] [ "." ] "2020.01.01"
```

I think that should be slightly faster to compute

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133762` at 35495aa0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.yt-dlp</li>
    <li>python38Packages.yt-dlp-light</li>
    <li>yt-dlp (python39Packages.yt-dlp)</li>
    <li>yt-dlp-light (python39Packages.yt-dlp-light)</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/tools/misc/yt-dlp/default.nix:51:0:
```
   |
51 |     description = "Command-line tool to download videos from YouTube.com and other sites (youtube-dl fork)";
   | ^
```
</li>
  </ul>
</details>

**Comment by mkg20001** [comment]
Both the upgrade and the 0 fix have been added to master in different PRs

Thanks for the PR anyways, it had some useful fixes that were copied.

**Pull request closed without merging**